magma/passes/insert_coreir_wires.py

```diff
@@ -25,8 +26,9 @@ def __init__(self, T):
         if issubclass(T, Digital):
             coreir_lib = "corebit"
             coreir_genargs = None
-            # Convert to bit so we wrap named types like clock if necessary
-            T = Bit
+            if issubclass(T, (AsyncReset, AsyncResetN, Clock)):
```

Comment:
what is the case where we don't want `T = Bit` ?

Reply:
ResetN or Reset (or any other subclass of Bit that's not AsyncReset/Clock).  The reason that the above types are special cased are that they correspond to coreir named types.  reset/resetn for example can just be treated as normal coreir bit